### PR TITLE
[rebased] Compare timebase of netplay users to detect desyncs.

### PR DESCRIPTION
**Pull request by comex**
Based on #2178.  Haven't done much more than rebase and compile check yet, since I'm about to go to sleep, but in the spirit of getting things done...

### CODE REVIEW

File: Source/Core/Core/NetPlayServer.cpp [hunk 1]
```diff
@@ -581,6 +583,50 @@ unsigned int NetPlayServer::OnData(sf::Packet& packet, Client& player)
 	}
 	break;
 
+	case NP_MSG_TIMEBASE:
+	{
+		u32 x, y, frame;
+		packet >> x;
+		packet >> y;
+		packet >> frame;
+		s_timebase[frame].emplace_back(x | ((u64)y >> 32), player.pid);
```
Comment (lioncash) [inline Source/Core/Core/NetPlayServer.cpp]: Shifting right by 32 would be incorrect here if I'm not mistaken.

**Comment by comex** [comment]
Okay, should actually work now.

File: Source/Core/Core/Core.cpp [hunk 2]
```diff
@@ -129,6 +130,13 @@ void SetIsFramelimiterTempDisabled(bool disable)
 std::string GetStateFileName() { return s_state_filename; }
 void SetStateFileName(const std::string& val) { s_state_filename = val; }
 
+void FrameUpdateOnCPUThread()
+{
+	if (!IsCPUThread()) abort();
```
Comment (lioncash) [inline Source/Core/Core/Core.cpp]: Uh

Comment (comex) [inline Source/Core/Core/Core.cpp]: Oops, that wasn't supposed to be left in.

Comment (comex) [inline Source/Core/Core/Core.cpp]: Fixed.

**Comment by JMC47** [comment]
In a quick test this is absolutely amazing.

**Comment by JMC47** [comment]
<JMC47> Lioncash, review 2517 ;D
<JMC47> it's pertinent to my interests.
<Lioncash> i dont have any issues with it.

**Pull request merged**